diff -ur cpptraj_orig/configure cpptraj/configure
--- cpptraj_orig/configure	2020-06-16 11:52:19.052789235 +0200
+++ cpptraj/configure	2020-06-16 11:54:09.537566064 +0200
@@ -2071,7 +2071,7 @@
 fi
 # CUDA
 if [ $USE_CUDA -eq 1 ] ; then
-  CUDA_TARGET='cuda_kernels/libcpptraj_cuda.a'
+  CUDA_TARGET='cuda_kernels/libcpptraj_cuda.a cuda_kernel_gist/lib_cuda_gist.a'
 fi
 # Readline
 if [ "${LIB_STAT[$LREADLINE]}" = 'bundled' ] ; then
Only in cpptraj: configure.orig
diff -ur cpptraj_orig/src/CMakeLists.txt cpptraj/src/CMakeLists.txt
--- cpptraj_orig/src/CMakeLists.txt	2020-06-16 11:52:19.076789403 +0200
+++ cpptraj/src/CMakeLists.txt	2020-06-16 11:54:09.537566064 +0200
@@ -1,4 +1,5 @@
 add_subdirectory(xdrfile)
+add_subdirectory(cuda_kernel_gist)
 
 #---------------------------------------------------------------------------------------------------------------------------------------------------------------------
 # Parse cpptrajfiles
diff -ur cpptraj_orig/src/Command.cpp cpptraj/src/Command.cpp
--- cpptraj_orig/src/Command.cpp	2020-06-16 11:52:19.084789460 +0200
+++ cpptraj/src/Command.cpp	2020-06-16 11:54:09.541566092 +0200
@@ -117,6 +117,7 @@
 #include "Action_Spam.h"
 #include "Action_Temperature.h"
 #include "Action_GIST.h"
+#include "Action_GIGIST.h"
 #include "Action_CreateReservoir.h"
 #include "Action_Density.h"
 #include "Action_PairDist.h"
@@ -317,6 +318,7 @@
   Command::AddCmd( new Action_FixAtomOrder(),  Cmd::ACT, 1, "fixatomorder" );
   Command::AddCmd( new Action_FixImagedBonds(),Cmd::ACT, 1, "fiximagedbonds" );
   Command::AddCmd( new Action_GIST(),          Cmd::ACT, 1, "gist" );
+  Command::AddCmd( new Action_GIGist(),        Cmd::ACT, 1, "gigist" );
   Command::AddCmd( new Action_GridFreeEnergy(),Cmd::ACT, 1, "gfe" ); // hidden
   Command::AddCmd( new Action_Grid(),          Cmd::ACT, 1, "grid" );
   Command::AddCmd( new Action_HydrogenBond(),  Cmd::ACT, 1, "hbond" );
Only in cpptraj/src: Command.cpp.orig
diff -ur cpptraj_orig/src/cpptrajfiles cpptraj/src/cpptrajfiles
--- cpptraj_orig/src/cpptrajfiles	2020-06-16 11:52:19.108789629 +0200
+++ cpptraj/src/cpptrajfiles	2020-06-16 11:54:09.541566092 +0200
@@ -38,6 +38,7 @@
         Action_FixAtomOrder.cpp \
         Action_FixImagedBonds.cpp \
         Action_GIST.cpp \
+        Action_GIGIST.cpp \
         Action_Grid.cpp \
         Action_GridFreeEnergy.cpp \
         Action_HydrogenBond.cpp \
diff -ur cpptraj_orig/src/Makefile cpptraj/src/Makefile
--- cpptraj_orig/src/Makefile	2020-06-16 11:52:19.092789517 +0200
+++ cpptraj/src/Makefile	2020-06-16 11:54:09.541566092 +0200
@@ -98,6 +98,9 @@
 cuda_kernels/libcpptraj_cuda.a: ../external.config.h
 	cd cuda_kernels && $(MAKE) all
 
+cuda_kernel_gist/lib_cuda_gist.a:
+	cd cuda_kernel_gist/ && $(MAKE) all
+
 # Dependency targets
 findDepend: FindDepend.o 
 	$(CXX) -o findDepend FindDepend.o
